### PR TITLE
Fix undo, redo and clear commands

### PR DESCRIPTION
**Pull request by GabrielYik**

### CODE REVIEW

File: src/main/java/seedu/planner/model/ModulePlanner.java [hunk 1]
```diff
@@ -103,14 +103,18 @@ public void setUserProfile(UserProfile u) {
...
+     * Returns a copy of all the {@code Semester}s.
      *
-     * @return An {@code ObservableList} containing all the {@code Semester}s
+     * @return A list of {@code Semester}s
      */
     @Override
-    public ObservableList<Semester> getSemesters() {
-        return FXCollections.unmodifiableObservableList(
-                FXCollections.observableList(semesters));
+    public List<Semester> getSemesters() {
```
Comment (Hilda-Ang) [inline src/main/java/seedu/planner/model/ModulePlanner.java]: Why change it to `List<...>`?

File: src/main/java/seedu/planner/model/semester/Semester.java [hunk 2]
```diff
@@ -89,10 +103,38 @@ public boolean containsModule(Module module) {
...
+    }
+
+    /**
+     * Copies the modules of {@code this} and
+     * returns them in an {@code ObservableList}.
+     *
+     * @return The copied modules of {@code this} wrapped
+     *  in an {@code ObservableList}
+     */
+    public ObservableList<Module> getModulesAsCopy() {
```
Comment (Hilda-Ang) [inline src/main/java/seedu/planner/model/semester/Semester.java]: Why do we need this method?

File: src/main/java/seedu/planner/model/ModulePlanner.java (hunk 1 above)
Comment (GabrielYik) [reply src/main/java/seedu/planner/model/ModulePlanner.java]: Cos got no need for semesters to be "observed"; they're not used in the UI directly. Only their underlying modules are used directly

File: src/main/java/seedu/planner/model/semester/Semester.java (hunk 2 above)
Comment (Hilda-Ang) [reply src/main/java/seedu/planner/model/semester/Semester.java]: Ohh okay got it.

File: src/main/java/seedu/planner/model/ModulePlanner.java (hunk 1 above)
Comment (Hilda-Ang) [reply src/main/java/seedu/planner/model/ModulePlanner.java]: Okay

**Pull request merged**